src/devTools.js

```diff
@@ -75,9 +77,10 @@ function transformAction(action, config) {
   return liftedAction;
 }
 
-export function send(action, state, options, type, instanceId) {
-  start(options);
-  setTimeout(() => {
+export function send(action, state, options, type, instanceId, hostnamePromise) {
+  setTimeout(async () => {
+    // makes sure the connection is established
+    await start(options, hostnamePromise);
```

Comment:
If the promise inside of the `start` function fails, the error is just being rethrown here(no need to handle), is this ok?

Reply:
I checked and `DeviceInfo.isEmulator` doesn't reject itself, but the promise we expect here is a little bit more than that and there's an `invariant` inside so I think maybe we should catch even if we only rethrow. Applied such logic in my newest changes.